services/azure-devops/azure-devops-tests.tester.js

```diff
@@ -174,11 +226,78 @@ t.create('test status with custom labels')
       skipped_label: 'n/a',
     },
   })
+  .intercept(nock =>
+    nock(azureDevOpsApiBaseUri)
+      .get(mockLatestBuildApiUriPath)
+      .reply(200, latestBuildResponse)
+      .get(mockTestResultSummaryApiUriPath)
+      .reply(200, mockTestResultSummaryResponse)
+  )
   .expectJSONTypes(
-    Joi.object().keys({ name: 'tests', value: isCustomAzureDevOpsTestTotals })
+    Joi.object().keys({
+      name: 'tests',
+      value: expectedCustomAzureDevOpsTestTotals,
+    })
   )
 
 t.create('test status with compact message and custom labels')
+  .get(mockBadgeUri, {
+    qs: {
+      compact_message: null,
+      passed_label: '💃',
+      failed_label: '🤦‍♀️',
+      skipped_label: '🤷',
+    },
+  })
+  .intercept(nock =>
+    nock(azureDevOpsApiBaseUri)
+      .get(mockLatestBuildApiUriPath)
+      .reply(200, latestBuildResponse)
+      .get(mockTestResultSummaryApiUriPath)
+      .reply(200, mockTestResultSummaryResponse)
```

Comment:
Are all of these the same? Could you move this up top? Something like

```js
const mockLatestBuildAndTestResultSummary = nock => nock(…)
```

Then you can use

```js
.intercept(mockLatestBuildAndTestResultSummary)
```

Reply:
Sure. Moved them up top in [`98093a`](https://github.com/badges/shields/pull/2454/commits/98093a372404b16535c46246d08558c8fab75e40).